src/BaseControllerV2.ts

```diff
@@ -174,11 +192,20 @@ export function getAnonymizedState<S extends Record<string, any>>(
 ): RecursivePartial<S> {
   return Object.keys(state).reduce((anonymizedState, _key) => {
     const key: keyof S = _key; // https://stackoverflow.com/questions/63893394/string-cannot-be-used-to-index-type-t
-    const metadataValue = metadata[key].anonymous;
-    if (isAnonymizingFunction(metadataValue)) {
-      anonymizedState[key] = metadataValue(state[key]);
-    } else if (metadataValue) {
-      anonymizedState[key] = state[key];
+    const propertyMetadata = metadata[key];
+    const propertyValue = state[key];
+    if (isStateMetadata(propertyMetadata)) {
+      if (isPrimitive(propertyValue) || !isNotArrayLike(propertyValue)) {
+        throw new Error('Invalid metadata');
+      }
+      anonymizedState[key] = getAnonymizedState(propertyValue, propertyMetadata);
```

Comment:
This line is where the error is:

Type 'RecursivePartial<S[keyof S]>' is not assignable to type 'S[keyof S] extends (infer U)[] ? RecursivePartial<U>[] : S[keyof S] extends Primitive ? S[keyof S] : RecursivePartial<S[keyof S]>'. ts(2322)

I tried to resolve this by adding type guards to "prove" that the first two conditions were not met.

Reply:
My guess is that it failed because TypeScript doesn't understand that this is the same _S[keyof S]_ as the other _S[keyof S]_? That is, these could be referring to two different properties that have different types. But it is the same property here, and I don't know how to tell TypeScript this.

Reply:
Well, I simplified the `RecursivePartial` type locally to test this theory and now I'm less sure. It looks like the Primitive type guard may not be working. I'm not 100% sure though.

Reply:
I got this working for now with a `// ts-ignore` comment :grimacing: